### PR TITLE
fixed header for mobile devices

### LINKED ISSUE #32: Mobile Responsive Header
Make the header mobile responsive. We use tailwindcss

### PR DESCRIPTION
**Pull request by chaitak-gorai**
## Description
- The header was not responsive for mobile devices. Solved this using tailwind.

## Solution:
- Added a Hamburger Button for mobile devices which when clicked expanded the nav-items menu.

## FIles Changed
- Header.vue (CSS only)

## Issues Solved
- Resolves #32 

## Preview (The issue)
mobile_header

## Preview After Solving
after1
after2
after3

### CODE REVIEW

File: resources/app/shared/includes/Header.vue [hunk 1]
```diff
@@ -1,380 +1,830 @@
...
+                    class="flex sm:flex-row flex-wrap p-2 md:items-center md:flex-row"
+                >
+                    <inertia-link
+                        :href="route('dashboard.index')"
+                        class="p-1 mx-2 border-transparent text-gray-400 rounded-full hover:text-white bg-gray-700 focus:outline-none focus:text-white focus:bg-gray-700"
+                    >
+                        <svg
+                            class="h-5 w-5"
+                            stroke="currentColor"
+                            xmlns="http://www.w3.org/2000/svg"
```
Comment (muarachmann) [inline resources/app/shared/includes/Header.vue]: Please use the `icon` component here for adding icons to the repo. We use https://heroicons.com for our reference.

File: resources/app/shared/includes/Header.vue [hunk 2]
```diff
@@ -1,380 +1,830 @@
...
+                        <button
+                            @click="isMenuOpen = !isMenuOpen"
+                            class="block lg:hidden text-white hover:text-gray-200 focus:text-gray-200 focus:outline-none"
+                        >
+                            <span class="inline-block align-middle">
+                                <svg
+                                    class="h-6 w-6 fill-current"
+                                    viewBox="0 0 24 24"
+                                    xmlns="http://www.w3.org/2000/svg"
+                                >
```
Comment (muarachmann) [inline resources/app/shared/includes/Header.vue]: Same here

**Review by muarachmann — CHANGES_REQUESTED** [review]

File: resources/app/shared/includes/Header.vue (hunk 1 above)
Comment (chaitak-gorai) [reply resources/app/shared/includes/Header.vue]: Sure, I am doing the required changes,

**Comment by chaitak-gorai** [comment]
## Update
- Hello @muarachmann , as requested, i have used the `icon` component for the SVG icons. Some icons were missing in the `Icon.vue` , so added them to the template

## FIles Changed
- Header.vue 
- Icon.vue

### Please review the code and suggest to me if some further changes are required.

**Comment by muarachmann** [comment]
This looks good just a few alignment with the `x` (close) button. @chaitak-gorai are you considering applying for GSoC?

**Comment by chaitak-gorai** [comment]
> This looks good just a few alignment with the `x` (close) button. @chaitak-gorai are you considering applying for GSoC?

Yes, I am applying for GSoC, and I find this project incredibly interesting. It's overwhelming to think about how my code could contribute to humanity through the health sector.

**Comment by muarachmann** [comment]
Thanks @chaitak-gorai while this menu is good. I will rather prefer a drawer menu with menu items. and submenus You can consider max level 2 for the sub-menus

**Comment by chaitak-gorai** [comment]
> Thanks @chaitak-gorai while this menu is good. I will rather prefer a drawer menu with menu items. and submenus You can consider max level 2 for the sub-menus

Ok.. I will modify the menu, and will use a drawer menu.

**Comment by muarachmann** [comment]
Any updates on this? @chaitak-gorai

**Comment by chaitak-gorai** [comment]
> Any updates on this? @chaitak-gorai

Hello @muarachmann, sorry for the delay, I was not available due to a family emergency. Will work on it now. I will show you the updated design by tomorrow.

**Comment by chaitak-gorai** [comment]
## Update
- Hello @muarachmann , as requested, i have updated the menu into a drawer menu. The drawer will slide from left to right when clicking the hamburger button. It has 2 levels for menus and submenus.

## FIles Changed
- tailwind.config: Added a tailwind component package `flowbite`
- Header.vue : Updated the UI
- Icon.vue : Added some new Icons

## Screenshot
updated

### Please look over the code and let me know if any more changes are required.

**Comment by muarachmann** [comment]
chevron should be normal without bg. Like on the homescreen. Aside that is there any reason why we had to pull in flowbite?
Also tapping out of the menu doesn't close the menu

**Comment by chaitak-gorai** [comment]
## Update

@muarachmann , I have made the requested icon update, and now the menu can be closed by tapping outside the screen.

We are utilizing flowbite, a component library built over tailwindcss, to implement interactive sections such as the drawer menu and submenus with improved design and style. This will not only enhance the UI but also accelerate the process of creating other UI sections during the migration from legacy code to the Laravel framework.

Additionally, please let me know if this update meets your requirements or if any further changes are needed.

## Screenshot
image

**Comment by muarachmann** [comment]
Nice @chaitak-gorai one last thing is the drawer when closed prevents interaction. Looks like the overlay is still there.

**Comment by chaitak-gorai** [comment]
Hey @muarachmann sorry but i couldn't find the issue on my end. 
I am attaching the screen. 
Can you please elaborate what is the possible issue here?.

https://user-images.githubusercontent.com/77141674/233722005-7b8530d6-5aca-4f59-ab6a-9e819cbec69b.mp4

 capture.

**Comment by muarachmann** [comment]
@chaitak-gorai tap on the body not the x to dismiss the modal, it dismisses but overlay is still there and prevents interaction

**Comment by chaitak-gorai** [comment]
> @chaitak-gorai tap on the body not the x to dismiss the modal, it dismisses but overlay is still there and prevents interaction

yeah got it thanks. :)
fixing it.

**Comment by chaitak-gorai** [comment]
## Fixed 

I have fixed the bug of the backdrop. Now the overlay doesn't exist after the drawer is closed by clicking outside it.

Please review it and suggest if any further change is required.

https://user-images.githubusercontent.com/77141674/233744579-169c668b-5af2-4e6c-b2cc-3f494ba6e350.mp4

**Comment by muarachmann** [comment]
LGMT

**Pull request merged**